src/hw1/vectors.py

```diff
@@ -0,0 +1,42 @@
+from math import sqrt, acos
+
+
+def check_vector(vector):
+    if not ((type(vector) == list) | (type(vector) == tuple)):
+        return (False, "Data should be stored in tuple of vector")
+    for x in vector:
+        if not ((type(x) == int) | (type(x) == float) | (type(x) == complex)):
+            return (False, "Vectors should consist of int, float of complex values")
+    return (True, "Ok")
+
+
+def check_multivector(*args):
+    for vector in args:
+        check_result, error_message = check_vector(vector)
+        if not check_result:
+            return (False, error_message)
+    return (True, "OK")
+
+
+def vector_check_exception_raiser(*args):
+    map(lambda x: check_vector(x), args)
+    (check_result, error_message) = check_multivector(*args)
+    if not check_result:
+        raise RuntimeError(error_message)
```

Comment:
This is way too cumbersome. Instead of returning `True, "OK"`, the functions above can just raise the error themselves. If everything is OK, there is no need to return anything.